src/cpp/src/sampler.hpp

```diff
@@ -57,6 +57,8 @@ class Sampler {
     std::mt19937 rng_engine;
     // { request_id, logit_processor }
     std::map<uint64_t, LogitProcessor> m_logit_processors;
+    // { request_id, { max_encoded_len, { stop_strings }}}
+    std::map<int64_t, std::pair<size_t, std::set<std::string>>> m_stop_strings;
```

Comment:
you need to clean this information in `clear_request_info` once request has finished / dropped / etc